### PR TITLE
[snapshot_download] upgrade to canonical separator

### PR DESCRIPTION
**Pull request by julien-c**
ping @severo

### CODE REVIEW

File: src/huggingface_hub/snapshot_download.py [hunk 1]
```diff
@@ -9,8 +9,9 @@
 from .utils import logging
 
 
-REPO_ID_SEPARATOR = "__"
-# ^ make sure this substring is not allowed in repo_ids on hf.co
+REPO_ID_SEPARATOR = "--"
+# ^ this substring is not allowed in repo_ids on hf.co
+# and is the canonical one we use for serialization of repo ids elsewhere.
```
Comment (julien-c) [inline src/huggingface_hub/snapshot_download.py:14]: (means that you could get the repo id from its snapshot path on disk, if ever needs be)

**Review by patrickvonplaten — APPROVED** [review]
Good for me! Think it will however force all repos to be downloaded & cached again. But should not be a big deal

**Review by LysandreJik — APPROVED** [review]
Ok LGTM!

**Comment by julien-c** [comment]
> Think it will however force all repos to be downloaded & cached again

yes but we already did that recently no? :)

but yeah i feel like now we should be at the final format so shouldn't change after that

**Comment by julien-c** [comment]
pinging @severo for a final review then let's merge!

**Review by severo — APPROVED** [review]
OK!

I put the moonlanding PR for reference: https://github.com/huggingface/moon-landing/pull/1657

**Pull request merged**